Sources/LiveKit/Support/StateSync.swift

```diff
@@ -15,75 +15,95 @@
  */
 
 import Foundation
+import Combine
+import os.lock
 
 internal typealias OnStateMutate<Value> = (_ state: Value, _ oldState: Value) -> Void
 
 @dynamicMemberLookup
 internal final class StateSync<Value> {
-
-    // use concurrent queue to allow multiple reads and block writes with barrier.
-    private let queue = DispatchQueue(label: "LiveKitSDK.state", qos: .default,
-                                      attributes: [.concurrent])
-
-    // actual value
-    private var _value: Value
+    
+    private let subject: CurrentValueSubject<Value, Never>
+    private let lock: UnsafeMutablePointer<os_unfair_lock_s>
+    
     public var onMutate: OnStateMutate<Value>?
+    
+    public var valuePublisher: AnyPublisher<Value, Never> {
```

Comment:
I think it will. One could use this publisher to subscribe to changes of the protected state - something like "onMutate". 
Though it's not used yet anywhere. So if you want me to remove it until it really is useful, that's totally ok.